### PR TITLE
[hubbard] Use enum type to set up the method for generating the hubbard subspace

### PR DESCRIPTION
**Pull request by mtaillefumier**
content of the  PR 
- regroups three different input parameters controlling how the hubbard subspace is generated in to one parameter only. The API already offers this to some extend.
- Adding atomic orbitals for the hubbard subspace through the Fortran interface assumes that the atomic type is fully initialized which is not the case when the potential information are stored in external files.

### CODE REVIEW

File: src/context/DOC.md
Comment (toxa81) [inline src/context/DOC.md:1]: This file should not be added, it is a by-product of the api generator script and still WIP

**Review by toxa81 — CHANGES_REQUESTED** [review]
DOC.md needs to be removed, otherwise it's fine

**Comment by toxa81** [comment]
> We may need a new version of sirius before I can open a PR on cp2k side of things

I fully agree, we are ready to tag 7.5

File: src/context/DOC.md
Comment (mtaillefumier) [reply src/context/DOC.md:1]: Ok I will fix it

Comment (mtaillefumier) [reply src/context/DOC.md:1]: removed.

**Pull request merged**